src/batou_ext/fcio.py

```diff
@@ -305,6 +309,69 @@ def get_diff(self, old, new):
         return result
 
 
+class DirectoryXMLRPC(batou.component.Component):
+    def configure(self):
+        self.xmlrpc = create_xmlrpc_client(self.environment)
+        self.provide("directory-xmlrpc", self.xmlrpc)
+
+
+class MaintenanceStart(batou.component.Component):
+    def configure(self):
+        self.require("needs-maintenance", strict=False, reverse=True)
+        self.xmlrpc = self.require_one("directory-xmlrpc")
+
+    def verify(self):
+        raise batou.UpdateNeeded()
+
+    def update(self):
+        change_maintenance_state(
+            self.xmlrpc, self.host.name, desired_state=True
+        )
+
+
+class MaintenanceEnd(batou.component.Component):
+    def configure(self):
+        self.require("needs-maintenance", strict=False)
+        self.xmlrpc = self.require_one("directory-xmlrpc")
+
+    def verify(self):
+        raise batou.UpdateNeeded()
```

Comment:
I still have bad feelings about this: every deployment with no other change will set the RG to maintenance for a short time with this.

Unfortunately I don't have a better idea after reading through batou's sources: when `MaintenanceStart` is deployed, we don't know yet if anything else will change later that needs maintenance. We could of course run all of the `verify()` methods of these components in here, but this doesn't take their sub-components into account.

Ideas? Or is this good enough?

Reply:
Can be modified now.

Reply:
See https://github.com/flyingcircusio/batou_ext/pull/213#discussion_r1952884922